### PR TITLE
Feature/44 segmentio tracking

### PR DESCRIPTION
**Pull request by bezidejni**
Added segment.io tracking to all the things we discussed, need a code review.

### CODE REVIEW

File: mediathread/templates/base.html [hunk 1]
```diff
@@ -7,10 +7,12 @@
     	<meta http-equiv="content-type" content="text/html; charset=utf-8" />
     	<title>Mediathread {% block title %}&mdash; Switch Course{% endblock %}</title>
 
+        {% if not settings.DEBUG %}
```
Comment (natea) [inline mediathread/templates/base.html:10]: What happens if this script isn't loaded on the pages where it's expecting the analytics lib to be loaded? does it just silently ignore those calls?

File: mediathread/templates/base.html [hunk 2]
```diff
@@ -247,7 +249,7 @@ <h4 id="logo_appsembler"></h4>
         <script type="text/javascript" src="{% static 'js/app/projects/projectlist.js' %}"></script>
     
         {% if not settings.DEBUG %}
-            <script>                
+            <script type="text/javascript">
                 {% if settings.GOOGLE_ANALYTICS_ID %}
                     var _gaq = _gaq || [];
                     _gaq.push(['_setAccount', '{{settings.GOOGLE_ANALYTICS_ID}}']);
```
Comment (natea) [inline mediathread/templates/base.html]: Do we even still need this if we're using segment.io for handling Google Analytics?

File: mediathread/user_accounts/views.py [hunk 3]
```diff
@@ -121,6 +116,13 @@ def form_valid(self, form):
                 email=email,
                 type="Student"
             )
+            analytics.identify(
+                email,
+                {
+                    "email": email,
+                    "type": "Student"
```
Comment (natea) [inline mediathread/user_accounts/views.py:123]: Where does this get set if it's an instructor and not a student?

Comment (bezidejni) [inline mediathread/user_accounts/views.py:123]: Well currently we only support inviting students to a course. When we add that list of people in the course where you can promote people to the faculty group, I'll add the call to segment.io which will identify that user as a instructor.

Comment (natea) [inline mediathread/user_accounts/views.py:123]: gotcha. that makes sense.

On Thu, Aug 15, 2013 at 12:50 PM, Filip Jukić notifications@github.comwrote:

> In mediathread/user_accounts/views.py:
> 
> > @@ -121,6 +116,13 @@ def form_valid(self, form):
> >                  email=email,
> >                  type="Student"
> >              )
> > -            analytics.identify(
> > -                email,
> > -                {
> > -                    "email": email,
> > -                    "type": "Student"
> 
> Well currently we only support inviting students to a course. When we add
> that list of people in the course where you can promote people to the
> faculty group, I'll add the call to segment.io which will identify that
> user as a instructor.
> 
> —
> Reply to this email directly or view it on GitHubhttps://github.com/appsembler/mediathread/pull/11/files#r5793802
> .

## 

nate@appsembler.com
+1 (617) 517-4953
http://twitter.com/natea | http://linkedin.com/in/natea

File: mediathread/templates/base.html (hunk 1 above)
Comment (bezidejni) [inline mediathread/templates/base.html:10]: I added a check to the identify call in base.html as well so no errors are thrown.

Comment (natea) [inline mediathread/templates/base.html:10]: ok, cool.

**Pull request merged**